Tests/UnitTests/Core/Detector/DetectorTests.cpp

```diff
@@ -112,6 +112,24 @@ BOOST_AUTO_TEST_CASE(DetectorConstruction) {
   BOOST_CHECK_EQUAL(
       det012, unpackToShared<const Acts::Experimental::Detector>(*det012));
 
+  // Check surface visiting
+  // Test the visitor pattern for surfaces
+  struct CountSurfaces {
+    unsigned int counter = 0;
+
+    void operator()(const Acts::Surface* s) {
+      if (s != nullptr) {
+        counter++;
+      }
+    }
+  };
```

Comment:
I was thinking, if we should add this to core as a helper, since we might want to check built geometries for consistency?
Same also for `struct ReachSurface` below.

Reply:
This is three lines for the moment, I don't think we need this counter currently. But if it pops up as a consistency_checker (good idea!) I will come back to this.